### PR TITLE
feature: add histogram

### PR DESCRIPTION
**Pull request by gcor**

### CODE REVIEW

File: components/Map/MainContainer/MapMainContainer.tsx [hunk 1]
```diff
@@ -30,64 +31,123 @@ interface Props {
...
-            <Copyright />
-        </>
+        <Histogram
+            data={data}
+            onChange={(value) => console.log(value)}
+            defaultMin={2018}
+            defaultMax={2021}
+        />
     );
+    // const position: [number, number] = COORDS_EKATERINBURG;
```
Comment (gcor) [inline components/Map/MainContainer/MapMainContainer.tsx:93]: comments just for demo

File: .eslintrc.json [hunk 2]
```diff
@@ -13,6 +13,9 @@
     "react/jsx-indent-props": ["error", 4],
     "react/jsx-props-no-spreading": "off",
     "react/jsx-filename-extension": [1, { "extensions": [".tsx"] }],
+    "react/require-default-props": "off",
```
Comment (gcor) [inline .eslintrc.json:16]: ts vs defaultprops conflict

File: components/Histogram/components/Slider.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+interface Props {
+    width: number;
+    min: number;
+    max: number;
+    onChange: Function;
+    currentMin?: number;
+    currentMax?: number;
+}
+
+export function Slider({ width, min, max, currentMin = min, currentMax = max, onChange }: Props) {
```
Comment (gcor) [inline components/Histogram/components/Slider.tsx:14]: inspired by random example from codesandbox

File: .eslintrc.json (hunk 2 above)
Comment (a-chabin) [reply .eslintrc.json:16]: not problem :)

File: components/Histogram/components/Slider.tsx (hunk 3 above)
Comment (a-chabin) [reply components/Histogram/components/Slider.tsx:14]: nice!

with keyboard support :)

File: components/Histogram/components/Slider.module.css [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+.thumb {
+    pointer-events: none;
+    position: absolute;
+    height: 0;
+}
+
+.thumb_zindex_3 {
+    z-index: 3;
```
Comment (a-chabin) [inline components/Histogram/components/Slider.module.css:37]: it seems to be ok to do just style, as for width :)

**Comment by a-chabin** [comment]
Can we increase clickable area for ranges?

https://user-images.githubusercontent.com/22644149/213459069-b5084860-8a03-4ade-803a-9b4a42fa1a10.mov

click across the entire width and 150% height

File: components/Histogram/components/Slider.module.css (hunk 4 above)
Comment (gcor) [reply components/Histogram/components/Slider.module.css:37]: removed

File: components/Histogram/components/Slider.module.css [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+.thumb_zindex_4 {
+    z-index: 4;
+}
+
+.thumb_zindex_5 {
+    z-index: 5;
+}
+
+/* For Chrome browsers */
+.thumb::-webkit-slider-thumb {
```
[the remark this answers is not in the document]
Comment (gcor) [reply components/Histogram/components/Slider.module.css:49]: lets use scale for focus too?

**Comment by gcor** [comment]
> Can we increase clickable area for ranges?
> 
>  Screen.Recording.2023-01-19.at.18.47.52.mov 
> click across the entire width and 150% height of line

now all histogram is clickable area :-)
desktop-only

File: components/Histogram/Histogram.tsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+interface Props {
+    data: HistogramData;
+    onChange: (range: Range) => void;
+    width?: number;
+    height?: number;
+    defaultMin?: number;
+    defaultMax?: number;
+}
+
+export function Histogram({
```
[the remark this answers is not in the document]
Comment (gcor) [reply components/Histogram/Histogram.tsx]: +

**Review by a-chabin — APPROVED** [review]

**Comment by a-chabin** [comment]
Thank you so much for your work! It is very cool!

**Pull request merged**